UnityProject/Assets/Scripts/Messages/Server/MetaDataLayerMessage.cs

```diff
@@ -0,0 +1,131 @@
+using System.Collections;
+using System.Collections.Generic;
+using Messages.Server;
+using Mirror;
+using Tilemaps.Behaviours.Layers;
+using UnityEngine;
+
+public class MetaDataLayerMessage : ServerMessage<MetaDataLayerMessage.NetMessage>
+{
+	public struct NetMessage : NetworkMessage
+	{
+		public List<DelayedData> Changes;
+		public uint MatrixSyncNetID;
+	}
+
+	//just a best guess, try increasing it until the message exceeds mirror's limit
+	private static readonly int MAX_CHANGES_PER_MESSAGE = 350;
+
+
+	public override void Process(NetMessage msg)
+	{
+		if (CustomNetworkManager.IsServer) return;
+		LoadNetworkObject(msg.MatrixSyncNetID);
+
+		var MetaDataLayer = NetworkObject.transform.parent.GetComponentInChildren<MetaDataLayer>();
+		foreach (var Change in msg.Changes)
+		{
+			MetaDataLayer.Get(Change.Position).IsSlippery = Change.IsSlippy;
+		}
+	}
+
+	public static void SendTo(GameObject managerSubject, NetworkConnection recipient,
+		Dictionary<Vector3Int, MetaDataNode> changeList)
+	{
+		if (changeList == null || changeList.Count == 0) return;
+		var netID = managerSubject.transform.parent.GetComponent<NetworkedMatrix>().MatrixSync.netId;
+		foreach (var changeChunk in changeList.Chunk(MAX_CHANGES_PER_MESSAGE))
+		{
+			List<DelayedData> Changes = new List<DelayedData>();
+
+			foreach (var metaData in changeChunk)
+			{
+				Changes.Add(new DelayedData()
+				{
+					Position = metaData.Key,
+					IsSlippy = metaData.Value.IsSlippery
+				});
+			}
+
+			NetMessage msg = new NetMessage
+			{
+				MatrixSyncNetID = netID,
+				Changes = Changes
+			};
+
+			SendTo(recipient, msg);
+		}
+	}
+
+	public static void Send(GameObject managerSubject, List<MetaDataNode> changeList)
+	{
+		var netID = managerSubject.transform.parent.GetComponent<NetworkedMatrix>().MatrixSync.netId;
+		// foreach (var changeChunk in changeList.Chunk(MAX_CHANGES_PER_MESSAGE)) //TODO Check it's not too big maybe
+		// {
+		List<DelayedData> Changes = new List<DelayedData>();
+
+		foreach (var metaData in changeList)
+		{
+			Changes.Add(new DelayedData()
+			{
+				Position = metaData.Position,
+				IsSlippy = metaData.IsSlippery
+			});
+		}
+
+		NetMessage msg = new NetMessage
+		{
+			MatrixSyncNetID = netID,
+			Changes = Changes
+		};
```

Comment:
This is duplicated code from `SendTo` above. Can you extract it into a method or similar?

Reply:
I can't really because one uses dictionary one uses list

Reply:
You could do it by taking a `IEnumerable<MetaDataNode>` as a parameter.
I noticed the commented out code that's actually present in the above method, which makes me want this to be factored out even more.

Reply:
You can get an `IEnumerable` from `dictionary.Values`.
Then both methods will be exactly the same except the `SendTo` and `SendToAll` and it can be factored out.